### PR TITLE
Conditional chaining

### PR DESCRIPTION
**Pull request by grouma**
Introduce a new flag --chain-stack-traces which providers users with a way to conditionally enable stack trace chaining. This will be useful for those users that want improved test performance at the cost of debuggability.

### CODE REVIEW

File: lib/src/backend/invoker.dart [hunk 1]
```diff
@@ -289,6 +289,11 @@ class Invoker {
     _controller.addError(error, stackTrace);
     removeAllOutstandingCallbacks();
 
+    if (!liveTest.test.metadata.chainStackTraces) {
+      _printsOnFailure.add("Consider enabling the flag chain-stack-traces to "
```
Comment (matanlurey) [inline lib/src/backend/invoker.dart]: Could we make this a little more specific, perhaps:

"Consider ...\n For example, pub run test --chain-stack-traces"

**Comment by matanlurey** [comment]
Not expecting for this PR, but any thought about auto-rerunning with chaining enabled when a test failed but keeping it off by default?

**Comment by kevmoo** [comment]
> Not expecting for this PR, but any thought about auto-rerunning with chaining enabled when a test failed but keeping it off by default?

@matanlurey – file a bug (feature request) ?

**Comment by matanlurey** [comment]
Filed https://github.com/dart-lang/test/issues/585.

File: lib/src/backend/invoker.dart (hunk 1 above)
Comment (grouma) [inline lib/src/backend/invoker.dart]: Done.

File: lib/src/runner/configuration/args.dart [hunk 2]
```diff
@@ -85,6 +85,12 @@ final ArgParser _parser = (() {
           'Implies --concurrency=1 and --timeout=none.\n'
           'Currently only supported for browser tests.',
       negatable: false);
+  parser.addFlag("chain-stack-traces",
+      help: 'Chains stack traces to provide greater exception details\n'
```
Comment (kevmoo) [inline lib/src/runner/configuration/args.dart]: `Chains` -> `Chained`?

Comment (grouma) [inline lib/src/runner/configuration/args.dart]: Done.

**Pull request merged**